### PR TITLE
feature/ED 3538 price ui indicator vomm

### PR DESCRIPTION
**Pull request by martinmcneela**
Ticket: https://edozohq.atlassian.net/browse/ED-3538

Why this is needed:

Link to Figma doc: https://www.figma.com/file/sUsVhNqXyq8zfVgT6n7qbR/Voom-Maps?node-id=235%3A5983

### CODE REVIEW

File: src/PriceIndicator/PriceIndicator.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  useEffectAfterMount(() => {
+    setIsAdditionVisible(true);
+    const timer = setTimeout(() => setIsAdditionVisible(false), 400);
+
+    return () => clearTimeout(timer);
+  }, [formattedAddition]);
+
+  return (
+    <React.Fragment>
+      {/* Tooltip UI */}
```
Comment (zac-robinson) [inline src/PriceIndicator/PriceIndicator.tsx]: Comment needed?

File: src/PriceIndicator/PriceIndicator.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      <AnimatePresence>
+        {isAdditionVisible && (
+          <PriceUpdateIndicator variants={updaterVariants} initial="initial" animate="animate" exit="exit">
+            + {formattedAddition}
+          </PriceUpdateIndicator>
+        )}
+      </AnimatePresence>
+      {/* Badge and hover UI */}
```
Comment (zac-robinson) [inline src/PriceIndicator/PriceIndicator.tsx]: Again comment?

File: src/PriceIndicator/PriceIndicator.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+    return () => clearTimeout(timer);
+  }, [formattedAddition]);
+
+  return (
+    <React.Fragment>
+      {/* Tooltip UI */}
+      <AnimatePresence>
+        {isAdditionVisible && (
+          <PriceUpdateIndicator variants={updaterVariants} initial="initial" animate="animate" exit="exit">
```
Comment (zac-robinson) [inline src/PriceIndicator/PriceIndicator.tsx]: Can you point me at docs saying what the initial prop does with value initial? I can't find it. Same with animate and exit?

Comment (zac-robinson) [reply src/PriceIndicator/PriceIndicator.tsx]: Oh do they relate to the variants you pass in?  Can we name them slightly differently then with like initialPosition, animatoTo and exitAnimation or something? Happy to talk about this one

File: src/PriceIndicator/PriceIndicator.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+<Meta title="Components|PriceIndicator" />
+
+# PriceIndicator
+
+## Description
+
+PriceIndicator is a simple styled and animated set of elements to be use to indicate users that the costs of an item (map) are increasing
+
+### Default
```
Comment (zac-robinson) [inline src/PriceIndicator/PriceIndicator.stories.mdx]: an we add a second story where active is passed in as `true`

File: package.json [hunk 5]
```diff
@@ -31,21 +31,23 @@
     "downshift": "^3.4.8"
   },
   "peerDependencies": {
-    "react": "^16.0.0",
-    "react-dom": "^16.0.0",
+    "react": "^16.12.0",
+    "react-dom": "^16.12.0",
```
Comment (zac-robinson) [inline package.json]: Are we going to need to upgrade the React version in the app if we are making 16.12 a peer dep?

File: src/index.tsx [hunk 6]
```diff
@@ -11,5 +11,7 @@ export * from './List';
 export { ToggleGroup } from './ToggleGroup';
 export { Switch } from './Switch';
 export { Badge } from './Badge';
+export { PriceIndicator } from './PriceIndicator';
+export * from './hooks';
```
[the remark this answers is not in the document]
Comment (martinmcneela) [reply src/index.tsx]: Figured we may need it and there would be no point duplicating the code in the app. Also won't add to the size anyway as we are using it.

File: src/PriceIndicator/PriceIndicator.tsx (hunk 1 above)
Comment (martinmcneela) [reply src/PriceIndicator/PriceIndicator.tsx]: I can remove it, I left it there so it was clear what was happening. You think it is clear without?

File: src/PriceIndicator/PriceIndicator.tsx (hunk 2 above)
Comment (martinmcneela) [reply src/PriceIndicator/PriceIndicator.tsx]: As above?

File: src/PriceIndicator/PriceIndicator.tsx (hunk 3 above)
Comment (martinmcneela) [reply src/PriceIndicator/PriceIndicator.tsx]: here you go: https://www.framer.com/api/motion/animation#variants

I think it is inferred that this is talking about the position? Have a look at the docs and if you still want it changed I can.

File: package.json (hunk 5 above)
Comment (martinmcneela) [reply package.json]: yup, I figured we can do this, give it a go and if the upgrade is causing issues in maps we can revert. But there should be no issues upgrading to 16.12.

File: src/index.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (zac-robinson) [reply src/index.tsx]: just feels a weird thing to have exported from a component lib to have a custom hook too

File: src/PriceIndicator/PriceIndicator.tsx (hunk 2 above)
Comment (zac-robinson) [reply src/PriceIndicator/PriceIndicator.tsx]: Can you change the comments to total indicator and addition indicator or something? I was confused by the name tooltip

File: package.json (hunk 5 above)
Comment (zac-robinson) [reply package.json]: yeah shouldnt be issues just need to make sure we do bump the maps version

File: src/PriceIndicator/PriceIndicator.tsx (hunk 3 above)
Comment (zac-robinson) [reply src/PriceIndicator/PriceIndicator.tsx]: Yeah seems a slightly odd API but happy for it to stay as is

File: src/index.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (martinmcneela) [reply src/index.tsx]: Not really most component libraries export helpers/utils.

File: src/PriceIndicator/PriceIndicator.tsx (hunk 1 above)
Comment (martinmcneela) [reply src/PriceIndicator/PriceIndicator.tsx]: removed it, second thought

File: package.json (hunk 5 above)
Comment (martinmcneela) [reply package.json]: https://edozohq.atlassian.net/browse/MAPS-96?atlOrigin=eyJpIjoiNWMzMTdjY2UzY2NhNGZmMWIyYmIxZmMyZTBkMDJkODIiLCJwIjoiaiJ9

File: src/PriceIndicator/PriceIndicator.tsx (hunk 2 above)
Comment (zac-robinson) [reply src/PriceIndicator/PriceIndicator.tsx]: The comments are still here

File: src/index.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (zac-robinson) [reply src/index.tsx]: It still feels strange to me. My thoughts are a hook isn't a component, so feels odd to me to be exported from a component lib, also its not a helper for this library, its just a custom hook that can be used in any component

**Comment by martinmcneela** [comment]
Disagree on the hooks export (if it is there, no issue, extra size exporting it for us to use in apps rather than duplicating the code) but I don't wanna keep going around in circles, so remove it.

**Pull request merged**